core/src/main/scala/chisel3/Element.scala

```diff
@@ -57,6 +57,16 @@ abstract class Element extends Data {
   override def litOption:                Option[BigInt] = litArgOption.map(_.num)
   private[chisel3] def litIsForcedWidth: Option[Boolean] = litArgOption.map(_.forcedWidth)
 
+  private[chisel3] def legacyAlwaysStrictConnect(that: Data)(implicit sourceInfo: SourceInfo): Unit = {
+    // If the source is a DontCare, generate a DefInvalid for the sink,
+    //  otherwise, issue a Connect.
+    if (that == DontCare) {
+      pushCommand(DefInvalid(sourceInfo, Node(this)))
+    } else {
+      pushCommand(Connect(sourceInfo, Node(this), that.ref))
+    }
```

Comment:
Why are you directly using `Node` here instead of `.lref` which includes error detection for visibility?

Reply:
changed to lref and tests still pass